vterm-module-make.el

```diff
@@ -2,28 +2,58 @@
 
 (require 'files)
 
-(defvar vterm-install-buffer-name " *Install vterm"
+(defcustom vterm-module-compilation-flags ""
+  "Flags given to CMake to compile vterm-module.
+
+Currently, the flags available are:
+
+`USE_SYSTEM_LIBVTERM'. Set it to `yes' to use the version of
+libvterm installed on your system.
+
+This string is given verbatim to CMake, so it has to have the
+correct syntax. An example of meaningful value for this variable
+is `-DUSE_SYSTEM_LIBVTERM=yes'."
+  :type 'string
+  :group 'vterm)
+
+(defvar vterm-install-buffer-name " *Install vterm* "
   "Name of the buffer used for compiling vterm-module.")
 
+(defun vterm-module--cmake-is-available ()
+  "Return t if cmake is available.
+CMake is needed to build vterm, here we check that we can find
+the executable."
+
+  (unless (executable-find "cmake")
+    (error "Vterm needs CMake to be compiled. Please, install CMake"))
+  t)
+
 ;;;###autoload
 (defun vterm-module-compile ()
-  "This function compiles the vterm-module."
+  "Compile vterm-module."
   (interactive)
-  (let ((default-directory
-          (file-name-directory (file-truename (locate-library "vterm"))))
-        (make-commands
-         "mkdir -p build; \
-          cd build; \
-          cmake \
-            -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-            ..; \
-          make"))
-    (unless (file-executable-p (concat default-directory "vterm-module.so"))
-      (let* ((buffer (get-buffer-create vterm-install-buffer-name)))
-        (pop-to-buffer vterm-install-buffer-name)
-        (if (zerop (call-process "sh" nil buffer t "-c" make-commands))
-            (message "Compilation of emacs-libvterm module succeeded")
-          (error "Compilation of emacs-libvterm module failed!"))))))
+  (when (vterm-module--cmake-is-available)
+    (let* ((vterm-directory
+            (shell-quote-argument
+             (file-name-directory (file-truename (locate-library "vterm")))))
+          (make-commands
```

Comment:
This is not properly indented

Reply:
This is how `emacs-lisp-mode` indents this block. How would you indent it?

Reply:
See https://github.com/akermu/emacs-libvterm/pull/261#discussion_r386833022. `make-commands` is not on the same column as `vterm-directory` in the let binding.

Reply:
> make-commands is not on the same column as vterm-directory in the let binding. 

Yes, I see it now.